### ISSUE TITLE
ext_authz_error after upgrade to 0.10.6

### ISSUE THREAD

**Issue by otherguy** [pomerium/pomerium#1520]
## What happened?

I'm running Pomerium on a Synology NAS as a Docker container. After updating the container to 0.10.6 (from 0.8.3), I get an empty `HTTP 500` response on all routes. The log shows `ext_authz_error`, but no warnings, errors or fatals.

## What did you expect to happen?

There shouldn't be any breaking changes causing this to happen.

## How'd it happen?

- fetched the new pomerium v0.10.6 image
- updated container to use the new image
- tried to connect to to a service proxied by pomerium

## What's your environment like?

- Pomerium version 0.10.6
- Docker version 18.09.8, build bfed4f5
- Linux 3.10.105 #25426 SMP Wed Jul 8 03:19:33 CST 2020 x86_64 GNU/Linux synology_avoton_415+

## What's your config.yaml?

```config.yaml
address: ":8443" # optional, default is 443
pomerium_debug: true # optional, default is false
service: "all" # optional, default is all
log_level: "info" # optional, default is debug

certificate_file: "/certs/fullchain.pem"
certificate_key_file: "/certs/privkey.pem"
certificate_authority_file: "/certs/*.int.example.com/ca.cer"

policy:
  - from: https://hass.int.example.com
    to: http://172.16.1.190:8123
    allow_public_unauthenticated_access: true
  - from: https://radarr.int.example.com
    to: http://localhost:8310
    allowed_users:
      - me@example.com
```

## Docker environment

```bash
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
AUTOCERT_DIR=/data/autocert
VIRTUAL_PORT=8443
VIRTUAL_HOST=*.int.example.com
VIRTUAL_PROTO=https
AUTHORIZE_INTERNAL_URL=localhost:8443
IDP_SERVICE_ACCOUNT=[REDACTED]
OVERRIDE_CERTIFICATE_NAME=*.int.example.com
ADMINISTRATORS=me@example.com
IDP_PROVIDER=google
IDP_PROVIDER_URL=https://accounts.google.com
IDP_CLIENT_ID=[REDACTED].apps.googleusercontent.com
IDP_CLIENT_SECRET=[REDACTED]
SHARED_SECRET=[REDACTED]
COOKIE_SECRET=[REDACTED]
AUTHENTICATE_SERVICE_URL=https://authenticate.int.example.com
AUTHORIZE_SERVICE_URL=https://authorize.int.example.com
AUTHENTICATE_INTERNAL_URL=localhost:844
```

## What did you see in the logs?

```logs
 9:21AM INF internal/grpc: grpc with insecure addr=127.0.0.1:5443
 9:21AM INF gRPC server started port=33404
 9:21AM INF HTTP server started port=44167
 9:21AM INF internal/grpc: grpc with insecure addr=127.0.0.1:5443
 9:21AM INF authenticate: updating options checksum=959517f86536b912
 9:21AM INF enabled authenticate service host=authenticate.int.example.com
 9:21AM INF internal/grpc: grpc with insecure addr=127.0.0.1:5443
 9:21AM INF enabled authorize service
 9:21AM INF authorize: updating options checksum=959517f86536b912
 9:21AM INF authorize: ecdsa public key PublicKey=LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUZrd0V3WUhLb1pJemowQ0FRWUlLb1pJemowREFRY0RRZ0FFbENqU1hqcXJPM05aME9OZTlDOFAwckV3blVmRgpKZ3Z2b3ZOK0V0dWZEZmE5Q083UmZJRWVHYkQ3UVhCREVsSmVQZm8yVHVsaFRZRU1ndzJJeTZNZmhnPT0KLS0tLS1FTkQgUFVCTElDIEtFWS0tLS0tCg==
 9:21AM WRN failed to read databroker CA file error="open : no such file or directory"
 9:21AM INF enabled cache service
 9:21AM INF internal/grpc: grpc with insecure addr=authorize.int.example.com:443
 9:21AM INF proxy: updating options checksum=9cea981a5f00153b
 9:21AM INF initializing directory groups service=identity_manager
 9:21AM INF get all service=databroker type=type.googleapis.com/directory.Group
 9:21AM INF initializing directory users service=identity_manager
 9:21AM INF get all service=databroker type=type.googleapis.com/directory.User
 9:21AM INF refreshing directory users service=identity_manager
 9:21AM INF syncing sessions service=identity_manager
 9:21AM INF sync record_version= server_version= service=databroker type=type.googleapis.com/session.Session
 9:21AM INF syncing directory groups service=identity_manager
 9:21AM INF syncing users service=identity_manager
 9:21AM INF sync record_version= server_version= service=databroker type=type.googleapis.com/user.User
 9:21AM INF sync record_version= server_version=fa4a09b3-a23c-4c3a-ab5f-1bdabeca07e7 service=databroker type=type.googleapis.com/directory.Group
 9:21AM INF starting control-plane gRPC server addr=127.0.0.1:33404
 9:21AM INF starting control-plane HTTP server addr=127.0.0.1:44167
 9:21AM INF syncing directory users service=identity_manager
 9:21AM INF starting type sync
 9:21AM INF sync record_version= server_version=fa4a09b3-a23c-4c3a-ab5f-1bdabeca07e7 service=databroker type=type.googleapis.com/directory.User
 9:21AM INF checking for existing cluster members cluster_url=127.0.0.1 service=memberlist
 9:21AM INF contacted nodes contacted=2 members=[{"Addr":"172.17.0.1","DCur":0,"DMax":0,"DMin":0,"Meta":null,"Name":"pomerium","PCur":2,"PMax":5,"PMin":1,"Port":7946,"State":0}] service=memberlist
 9:21AM INF set id=03bj1y383ptkyjw service=databroker type=type.googleapis.com/directory.Group
 9:21AM INF set id=01hmsyys0jdmnu6 service=databroker type=type.googleapis.com/directory.Group
 9:21AM INF set id=google/102962863207122022968 service=databroker type=type.googleapis.com/directory.User
 9:21AM INF set id=google/C00qtjav9 service=databroker type=type.googleapis.com/directory.User
 9:21AM INF initializing epoch 0 (base id=0, hot restart version=disabled) name=main service=envoy
 9:21AM INF statically linked extensions: name=main service=envoy
 9:21AM INF   envoy.compression.decompressor: envoy.compression.gzip.decompressor name=main service=envoy
 9:21AM INF   envoy.dubbo_proxy.filters: envoy.filters.dubbo.router name=main service=envoy
 9:21AM INF   envoy.filters.listener: envoy.filters.listener.http_inspector, envoy.filters.listener.original_dst, envoy.filters.listener.original_src, envoy.filters.listener.proxy_protocol, envoy.filters.listener.tls_inspector, envoy.listener.http_inspector, envoy.listener.original_dst, envoy.listener.original_src, envoy.listener.proxy_protocol, envoy.listener.tls_inspector name=main service=envoy
 9:21AM INF   envoy.grpc_credentials: envoy.grpc_credentials.aws_iam, envoy.grpc_credentials.default, envoy.grpc_credentials.file_based_metadata name=main service=envoy
 9:21AM INF   envoy.thrift_proxy.transports: auto, framed, header, unframed name=main service=envoy
 9:21AM INF   envoy.thrift_proxy.filters: envoy.filters.thrift.rate_limit, envoy.filters.thrift.router name=main service=envoy
 9:21AM INF   envoy.udp_listeners: raw_udp_listener name=main service=envoy
 9:21AM INF   envoy.transport_sockets.upstream: envoy.transport_sockets.alts, envoy.transport_sockets.quic, envoy.transport_sockets.raw_buffer, envoy.transport_sockets.tap, envoy.transport_sockets.tls, raw_buffer, tls name=main service=envoy
 9:21AM INF   envoy.transport_sockets.downstream: envoy.transport_sockets.alts, envoy.transport_sockets.quic, envoy.transport_sockets.raw_buffer, envoy.transport_sockets.tap, envoy.transport_sockets.tls, raw_buffer, tls name=main service=envoy
 9:21AM INF   envoy.retry_host_predicates: envoy.retry_host_predicates.omit_canary_hosts, envoy.retry_host_predicates.omit_host_metadata, envoy.retry_host_predicates.previous_hosts name=main service=envoy
 9:21AM INF   envoy.dubbo_proxy.route_matchers: default name=main service=envoy
 9:21AM INF   envoy.stats_sinks: envoy.dog_statsd, envoy.metrics_service, envoy.stat_sinks.dog_statsd, envoy.stat_sinks.hystrix, envoy.stat_sinks.metrics_service, envoy.stat_sinks.statsd, envoy.statsd name=main service=envoy
 9:21AM INF   envoy.filters.udp_listener: envoy.filters.udp.dns_filter, envoy.filters.udp_listener.udp_proxy name=main service=envoy
 9:21AM INF   envoy.clusters: envoy.cluster.eds, envoy.cluster.logical_dns, envoy.cluster.original_dst, envoy.cluster.static, envoy.cluster.strict_dns, envoy.clusters.aggregate, envoy.clusters.dynamic_forward_proxy, envoy.clusters.redis name=main service=envoy
 9:21AM INF   envoy.resource_monitors: envoy.resource_monitors.fixed_heap, envoy.resource_monitors.injected_resource name=main service=envoy
 9:21AM INF   envoy.health_checkers: envoy.health_checkers.redis name=main service=envoy
 9:21AM INF   envoy.tracers: envoy.dynamic.ot, envoy.lightstep, envoy.tracers.datadog, envoy.tracers.dynamic_ot, envoy.tracers.lightstep, envoy.tracers.opencensus, envoy.tracers.xray, envoy.tracers.zipkin, envoy.zipkin name=main service=envoy
 9:21AM INF   envoy.dubbo_proxy.serializers: dubbo.hessian2 name=main service=envoy
 9:21AM INF   envoy.access_loggers: envoy.access_loggers.file, envoy.access_loggers.http_grpc, envoy.access_loggers.tcp_grpc, envoy.file_access_log, envoy.http_grpc_access_log, envoy.tcp_grpc_access_log name=main service=envoy
 9:21AM INF   envoy.retry_priorities: envoy.retry_priorities.previous_priorities name=main service=envoy
 9:21AM INF   envoy.compression.compressor: envoy.compression.gzip.compressor name=main service=envoy
 9:21AM INF   envoy.dubbo_proxy.protocols: dubbo name=main service=envoy
 9:21AM INF   http_cache_factory: envoy.extensions.http.cache.simple name=main service=envoy
 9:21AM INF   envoy.internal_redirect_predicates: envoy.internal_redirect_predicates.allow_listed_routes, envoy.internal_redirect_predicates.previous_routes, envoy.internal_redirect_predicates.safe_cross_scheme name=main service=envoy
 9:21AM INF   envoy.filters.network: envoy.client_ssl_auth, envoy.echo, envoy.ext_authz, envoy.filters.network.client_ssl_auth, envoy.filters.network.direct_response, envoy.filters.network.dubbo_proxy, envoy.filters.network.echo, envoy.filters.network.ext_authz, envoy.filters.network.http_connection_manager, envoy.filters.network.kafka_broker, envoy.filters.network.local_ratelimit, envoy.filters.network.mongo_proxy, envoy.filters.network.mysql_proxy, envoy.filters.network.postgres_proxy, envoy.filters.network.ratelimit, envoy.filters.network.rbac, envoy.filters.network.redis_proxy, envoy.filters.network.rocketmq_proxy, envoy.filters.network.sni_cluster, envoy.filters.network.sni_dynamic_forward_proxy, envoy.filters.network.tcp_proxy, envoy.filters.network.thrift_proxy, envoy.filters.network.zookeeper_proxy, envoy.http_connection_manager, envoy.mongo_proxy, envoy.ratelimit, envoy.redis_proxy, envoy.tcp_proxy name=main service=envoy
 9:21AM INF   envoy.thrift_proxy.protocols: auto, binary, binary/non-strict, compact, twitter name=main service=envoy
 9:21AM INF   envoy.filters.http: envoy.buffer, envoy.cors, envoy.csrf, envoy.ext_authz, envoy.fault, envoy.filters.http.adaptive_concurrency, envoy.filters.http.admission_control, envoy.filters.http.aws_lambda, envoy.filters.http.aws_request_signing, envoy.filters.http.buffer, envoy.filters.http.cache, envoy.filters.http.compressor, envoy.filters.http.cors, envoy.filters.http.csrf, envoy.filters.http.decompressor, envoy.filters.http.dynamic_forward_proxy, envoy.filters.http.dynamo, envoy.filters.http.ext_authz, envoy.filters.http.fault, envoy.filters.http.grpc_http1_bridge, envoy.filters.http.grpc_http1_reverse_bridge, envoy.filters.http.grpc_json_transcoder, envoy.filters.http.grpc_stats, envoy.filters.http.grpc_web, envoy.filters.http.gzip, envoy.filters.http.header_to_metadata, envoy.filters.http.health_check, envoy.filters.http.ip_tagging, envoy.filters.http.jwt_authn, envoy.filters.http.lua, envoy.filters.http.on_demand, envoy.filters.http.original_src, envoy.filters.http.ratelimit, envoy.filters.http.rbac, envoy.filters.http.router, envoy.filters.http.squash, envoy.filters.http.tap, envoy.grpc_http1_bridge, envoy.grpc_json_transcoder, envoy.grpc_web, envoy.gzip, envoy.health_check, envoy.http_dynamo_filter, envoy.ip_tagging, envoy.lua, envoy.rate_limit, envoy.router, envoy.squash name=main service=envoy
 9:21AM INF   envoy.upstreams: envoy.filters.connection_pools.http.generic, envoy.filters.connection_pools.http.http, envoy.filters.connection_pools.http.tcp name=main service=envoy
 9:21AM INF   envoy.resolvers: envoy.ip name=main service=envoy
 9:21AM INF HTTP header map info: name=main service=envoy
 9:21AM INF   request header map: 496 bytes: :authority,:method,:path,:protocol,:scheme,accept,accept-encoding,access-control-request-method,authorization,cache-control,connection,content-encoding,content-length,content-type,expect,grpc-accept-encoding,grpc-timeout,keep-alive,origin,proxy-connection,referer,te,transfer-encoding,upgrade,user-agent,via,x-client-trace-id,x-envoy-attempt-count,x-envoy-decorator-operation,x-envoy-downstream-service-cluster,x-envoy-downstream-service-node,x-envoy-expected-rq-timeout-ms,x-envoy-external-address,x-envoy-force-trace,x-envoy-hedge-on-per-try-timeout,x-envoy-internal,x-envoy-ip-tags,x-envoy-max-retries,x-envoy-original-path,x-envoy-original-url,x-envoy-retriable-header-names,x-envoy-retriable-status-codes,x-envoy-retry-grpc-on,x-envoy-retry-on,x-envoy-upstream-alt-stat-name,x-envoy-upstream-rq-per-try-timeout-ms,x-envoy-upstream-rq-timeout-alt-response,x-envoy-upstream-rq-timeout-ms,x-forwarded-client-cert,x-forwarded-for,x-forwarded-proto,x-ot-span-context,x-request-id name=main service=envoy
 9:21AM INF   request trailer map: 72 bytes:  name=main service=envoy
 9:21AM INF   response header map: 352 bytes: :status,access-control-allow-credentials,access-control-allow-headers,access-control-allow-methods,access-control-allow-origin,access-control-expose-headers,access-control-max-age,cache-control,connection,content-encoding,content-length,content-type,date,etag,grpc-message,grpc-status,keep-alive,location,proxy-connection,referer,server,transfer-encoding,upgrade,vary,via,x-envoy-attempt-count,x-envoy-decorator-operation,x-envoy-degraded,x-envoy-immediate-health-check-fail,x-envoy-ratelimited,x-envoy-upstream-canary,x-envoy-upstream-healthchecked-cluster,x-envoy-upstream-service-time,x-request-id name=main service=envoy
 9:21AM INF   response trailer map: 96 bytes: grpc-message,grpc-status name=main service=envoy
 9:21AM INF admin address: 127.0.0.1:9901 name=main service=envoy
 9:21AM INF runtime: layers:
   - name: base
     static_layer:
       {}
   - name: admin
     admin_layer:
       {} name=main service=envoy
 9:21AM INF loading tracing configuration name=config service=envoy
 9:21AM INF loading 0 static secret(s) name=config service=envoy
 9:21AM INF loading 1 cluster(s) name=config service=envoy
 9:21AM INF loading 0 listener(s) name=config service=envoy
 9:21AM INF loading stats sink configuration name=config service=envoy
 9:21AM INF RTDS has finished initialization name=runtime service=envoy
 9:21AM INF cm init: initializing cds name=upstream service=envoy
 9:21AM INF starting main dispatch loop name=main service=envoy
 9:21AM INF cds: add 10 cluster(s), remove 0 cluster(s) name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'pomerium-control-plane-http\'" name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'authorize.int.example.com\'" name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'policy-782e26835d1c8cd0\'" name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'policy-eca2b959e19046c3\'" name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'policy-2d5d64432963938\'" name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'policy-9ae50e424af29684\'" name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'policy-bf906ef4c30c9dd4\'" name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'policy-ac791ba43e788ce4\'" name=upstream service=envoy
 9:21AM INF "cds: add/update cluster \'policy-e3746bf47372f595\'" name=upstream service=envoy
 9:21AM INF cm init: all clusters initialized name=upstream service=envoy
 9:21AM INF all clusters initialized. initializing init manager name=main service=envoy
 9:21AM INF "lds: add/update listener \'https-ingress\'" name=upstream service=envoy
 9:21AM INF "lds: add/update listener \'grpc-ingress\'" name=upstream service=envoy
 9:21AM INF all dependencies initialized. starting workers name=config service=envoy
 9:21AM INF sync record_version= server_version= service=databroker type=type.googleapis.com/pomerium.config.Config
 9:21AM INF sync types service=databroker
 9:21AM INF starting data initial load type_url=type.googleapis.com/user.User
 9:21AM INF starting data initial load type_url=type.googleapis.com/directory.User
 9:21AM INF starting data initial load type_url=server_version
 9:21AM INF starting data initial load type_url=type.googleapis.com/pomerium.config.Config
 9:21AM INF starting data initial load type_url=type.googleapis.com/directory.Group
 9:21AM INF starting data initial load type_url=type.googleapis.com/session.Session
 9:21AM INF get all service=databroker type=type.googleapis.com/user.User
 9:21AM INF get all service=databroker type=server_version
 9:21AM INF get all service=databroker type=type.googleapis.com/directory.User
 9:21AM INF get all service=databroker type=type.googleapis.com/directory.Group
 9:21AM INF get all service=databroker type=type.googleapis.com/pomerium.config.Config
 9:21AM INF get all service=databroker type=type.googleapis.com/session.Session
 9:21AM INF starting data syncer type_url=type.googleapis.com/user.User
 9:21AM INF starting data syncer type_url=type.googleapis.com/session.Session
 9:21AM INF starting data syncer type_url=type.googleapis.com/pomerium.config.Config
 9:21AM INF starting data syncer type_url=server_version
 9:21AM INF starting data syncer type_url=type.googleapis.com/directory.Group
 9:21AM INF starting data syncer type_url=type.googleapis.com/directory.User
 9:21AM INF sync record_version= server_version=fa4a09b3-a23c-4c3a-ab5f-1bdabeca07e7 service=databroker type=type.googleapis.com/user.User
 9:21AM INF sync record_version= server_version=fa4a09b3-a23c-4c3a-ab5f-1bdabeca07e7 service=databroker type=type.googleapis.com/pomerium.config.Config
 9:21AM INF sync record_version=000000000002 server_version=fa4a09b3-a23c-4c3a-ab5f-1bdabeca07e7 service=databroker type=type.googleapis.com/directory.User
 9:21AM INF sync record_version=000000000002 server_version=fa4a09b3-a23c-4c3a-ab5f-1bdabeca07e7 service=databroker type=type.googleapis.com/directory.Group
 9:21AM INF sync record_version= server_version=fa4a09b3-a23c-4c3a-ab5f-1bdabeca07e7 service=databroker type=type.googleapis.com/session.Session
 9:21AM INF sync record_version=000000000001 server_version=fa4a09b3-a23c-4c3a-ab5f-1bdabeca07e7 service=databroker type=server_version
 9:23AM INF http-request authority=hass.int.example.com duration=22.664737 forwarded-for=46.124.171.199 method=GET path=/ referer= request-id=e7306e92-9a44-4303-9c30-cd0bdd105213 response-code=500 response-code-details=ext_authz_error service=envoy size=0 upstream-cluster= user-agent="Paw/3.2 (Macintosh; OS X/10.15.7) GCDHTTPRequest"
 9:24AM INF http-request authority=radarr.int.example.com duration=12.830617 forwarded-for=46.124.171.199 method=GET path=/ referer= request-id=2ed8eea7-aef7-4129-bc4b-dcc67432b337 response-code=500 response-code-details=ext_authz_error service=envoy size=0 upstream-cluster= user-agent="Paw/3.2 (Macintosh; OS X/10.15.7) GCDHTTPRequest"
```

## Additional context

Enabling debug gives me this:

```logs
 9:42AM DBG [C9] closing data_to_write=128 type=2 name=connection service=envoy
 9:42AM DBG [C9] setting delayed close timer with timeout 1000 ms name=connection service=envoy
 9:42AM DBG [C9] write flush complete name=connection service=envoy
 9:42AM DBG [C11] handshake error: 1 name=connection service=envoy
 9:42AM DBG [C11] TLS error: 268436504:SSL routines:OPENSSL_internal:TLSV1_ALERT_UNKNOWN_CA name=connection service=envoy
 9:42AM DBG [C11] closing socket: 0 name=connection service=envoy
 9:42AM DBG [C11] adding to cleanup list name=conn_handler service=envoy
 9:42AM DBG [C9] remote early close name=connection service=envoy
 9:42AM DBG [C9] closing socket: 0 name=connection service=envoy
 9:42AM DBG [C9] SSL shutdown: rc=0 name=connection service=envoy
 9:42AM DBG [C9]  name=connection service=envoy
 9:42AM DBG [C9] adding to cleanup list name=conn_handler service=envoy
 9:42AM DBG "[C0][S1479551408804713033] cluster \'pomerium-control-plane-grpc\' match for URL \'/envoy.service.accesslog.v2.AccessLogService/StreamAccessLogs\'" name=router service=envoy
 9:42AM DBG "[C0][S1479551408804713033] router decoding headers:\n\':method\', \'POST\'\n\':path\', \'/envoy.service.accesslog.v2.AccessLogService/StreamAccessLogs\'\n\':authority\', \'pomerium-control-plane-grpc\'\n\':scheme\', \'http\'\n\'te\', \'trailers\'\n\'content-type\', \'application/grpc\'\n\'x-envoy-internal\', \'true\'\n\'x-forwarded-for\', \'172.16.1.200\'" name=router service=envoy
 9:42AM ???  name=envoy service=envoy
 9:42AM DBG queueing request due to no available connections name=pool service=envoy
 9:42AM DBG creating a new connection name=pool service=envoy
 9:42AM DBG [C12] connecting name=client service=envoy
 9:42AM DBG [C12] connecting to 127.0.0.1:58039 name=connection service=envoy
 9:42AM DBG [C12] connection in progress name=connection service=envoy
 9:42AM DBG [C12] updating connection-level initial window size to 268435456 name=http2 service=envoy
 9:42AM DBG [C12] connected name=connection service=envoy
 9:42AM DBG [C12] connected name=client service=envoy
 9:42AM DBG [C12] attaching to next request name=pool service=envoy
 9:42AM DBG [C12] creating stream name=pool service=envoy
 9:42AM DBG [C0][S1479551408804713033] pool ready name=router service=envoy
 9:42AM INF http-request authority=radarr.int.example.com duration=18.290898 forwarded-for=46.124.171.199 method=GET path=/ referer= request-id=ab91ae6a-3cfc-4cf7-946d-4f7d4381596a response-code=500 response-code-details=ext_authz_error service=envoy size=0 upstream-cluster= user-agent="Paw/3.2 (Macintosh; OS X/10.15.7) GCDHTTPRequest"
 9:42AM DBG flushing stats name=main service=envoy
 9:42AM DBG tls inspector: new connection accepted name=filter service=envoy
 9:42AM DBG [C13] new connection name=conn_handler service=envoy
 9:42AM DBG [C13] handshake error: 1 name=connection service=envoy
 9:42AM DBG [C13] TLS error: 268435703:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER name=connection service=envoy
```

The `TLS error: 268435703:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER` seems to be the culprit. But I'm not sure why it's happening and why it only shows up as a debug log line, when it has such an impact.

**Comment by desimone** [comment]
Hey @otherguy !

Thank you for your detailed bug report. 

A few things jump out at me.

```
 9:21AM WRN failed to read databroker CA file error="open : no such file or directory"
```


```
certificate_file: "/certs/fullchain.pem"
certificate_key_file: "/certs/privkey.pem"
certificate_authority_file: "/certs/*.int.example.com/ca.cer"
```

```
SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
```

```
VIRTUAL_PORT=8443
VIRTUAL_HOST=*.int.example.com
VIRTUAL_PROTO=https
```

I'm a little confused by the interplay of the certificate settings `certificate_file*` / `auto_cert` / and `SSL_CERT_FILE` as well as the `VIRTUAL_` settings. 

Some other ideas, some of the old internal-url's are no longer required. 

```diff
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
AUTOCERT_DIR=/data/autocert
VIRTUAL_PORT=8443
VIRTUAL_HOST=*.int.example.com
VIRTUAL_PROTO=https
+#AUTHORIZE_INTERNAL_URL=localhost:8443
IDP_SERVICE_ACCOUNT=[REDACTED]
OVERRIDE_CERTIFICATE_NAME=*.int.example.com
ADMINISTRATORS=me@example.com
IDP_PROVIDER=google
IDP_PROVIDER_URL=https://accounts.google.com
IDP_CLIENT_ID=[REDACTED].apps.googleusercontent.com
IDP_CLIENT_SECRET=[REDACTED]
SHARED_SECRET=[REDACTED]
COOKIE_SECRET=[REDACTED]
AUTHENTICATE_SERVICE_URL=https://authenticate.int.example.com
+#AUTHORIZE_SERVICE_URL=https://authorize.int.example.com
+#AUTHENTICATE_INTERNAL_URL=localhost:844
```

**Comment by otherguy** [comment]
@desimone thank you for your help!

I followed your advice and removed `AUTHORIZE_INTERNAL_URL`, `AUTHORIZE_SERVICE_URL` and `AUTHENTICATE_INTERNAL_URL` as well as `AUTOCERT_DIR`, as it is not needed -- like you mentioned.

After a container restart, everything works as expected, which, ironically, I did not expect!

This line still shows up and I assume it's because there is no `/etc/ssl/certs/ca-certificates.crt` in the new pomerium Docker container. I might mount my local `ca-certificates.crt` into the container or remove it entirely, but currently everything is working just fine.

`WRN failed to read databroker CA file error="open : no such file or directory"`

Thank you for your help and I guess I will close this ticket!
